src/utils/DependsOnManager.ts

```diff
@@ -36,6 +36,22 @@ export class DependsOnManager {
     $$(this.facet.element).toggleClass('coveo-hidden', !this.parentFacetHasSelectedValues);
   }
 
+  private get dependentFacets() {
+    const allFacets = ComponentsTypes.getAllFacetsInstance(this.facet.root);
+    return allFacets.filter(facet => {
+      return this.facet.id === facet.options.dependsOn;
+    });
+  }
+
+  public get dependentFacetsHaveSelectedValues() {
+    const dependentFacetsWithValues = this.dependentFacets.filter(facet => {
+      const facetId = QueryStateModel.getFacetId(facet.options.id);
+      return this.valuesExistForFacetWithId(facetId);
+    });
+
+    return !!dependentFacetsWithValues.length;
```

Comment:
I think you could leverage `.some` since you need "one" of them to match:

```
return this.dependentFacets.some(facet => {
      const facetId = QueryStateModel.getFacetId(facet.options.id);
      return this.valuesExistForFacetWithId(facetId);
  });
```